### PR TITLE
Handle to_field_name correctly

### PR DESCRIPTION
**Pull request by njoyard**
This PR allows handling the `to_field_name` parameter correctly to use something other than `pk` as a value on `AgnocompleteModel`.

It also introduces a new `label()` method on `AgnocompleteModel` that is called by `item()`. Overriding it allows customizing item labels without interfering with how values are computed.

### CODE REVIEW

**Comment by njoyard** [comment]
I'm not certain to have added tests for everything that was needed. Waiting for review from @brunobord :)

**Review by brunobord — DISMISSED** [review]
Please add a paragraph and a sample in the documentation showing how we support "to_field_name" for non-pk fields + an item in the ChangeLog

**Review by brunobord — APPROVED** [review]

**Pull request merged**
